### ISSUE TITLE
[🐛 Bug]: RetryRequest -> unlimited retries?

### ISSUE THREAD

**Issue by ogolovanov** [SeleniumHQ/selenium#10437]
### What happened?

Hi.

I configured RemoteWebDriver with:
1) socks5 proxy server
2) pageLoadTimeout = 180s.

Everything is ok, but one of my driver->get() request "stalled" due to "problems" with proxy server.
And it seems selenium tries to execute that request indefinitely. ( RetryRequest class )
This is not what i wanted after setting pageLoadTimeout to 180s.
I expected to get TimeoutException from webdriver after 180s.

Question: how to avoid that unexpected behaviour?

Logs attached.

### How can we reproduce the issue?

```shell
Not possible to reproduce due to problems related to specific proxy server.
```

### Relevant log output

```shell
2022-03-11 11:24:07,360	org.openqa.selenium.remote.http.RetryRequest	Failure due to server error #1. Retrying.
2022-03-11 11:27:09,366	org.openqa.selenium.remote.http.RetryRequest	Failure due to server error #2. Retrying.
2022-03-11 11:30:11,367	org.openqa.selenium.remote.http.RetryRequest	Read timeout #3. Retrying.
2022-03-11 11:33:15,368	org.openqa.selenium.remote.http.RetryRequest	Read timeout #4. Retrying.
2022-03-11 11:36:19,369	org.openqa.selenium.remote.http.RetryRequest	Read timeout #5. Retrying.
2022-03-11 11:39:20,371	org.openqa.selenium.remote.http.RetryRequest	Failure due to server error #1. Retrying.
2022-03-11 11:42:21,372	org.openqa.selenium.remote.http.RetryRequest	Failure due to server error #1. Retrying.
2022-03-11 11:45:23,372	org.openqa.selenium.remote.http.RetryRequest	Failure due to server error #2. Retrying.
2022-03-11 11:48:25,373	org.openqa.selenium.remote.http.RetryRequest	Read timeout #3. Retrying.
2022-03-11 11:51:29,374	org.openqa.selenium.remote.http.RetryRequest	Read timeout #4. Retrying.
2022-03-11 11:54:33,374	org.openqa.selenium.remote.http.RetryRequest	Read timeout #5. Retrying.
2022-03-11 11:57:35,375	org.openqa.selenium.remote.http.RetryRequest	Failure due to server error #2. Retrying.
```

### Operating System

Ubuntu

### Selenium version

4.1.2

### What are the browser(s) and version(s) where you see this issue?

Chrome 98

### What are the browser driver(s) and version(s) where you see this issue?

ChromeDriver 98.0.4758.102

### Are you using Selenium Grid?

4.1.2

**Comment by pujagani** [comment]
Thank you for sharing the details. I was able to reproduce this locally by running a RemoteWebDriver against the Grid and the Grid server was not actually started. I saw similar logs as shared by you. The issue also only occurs if a single retry policy is used. I will be triaging this further to find the root cause.

**Comment by titusfortner** [comment]
@pujagani if server isn't started, wouldn't the failure happen when starting the session instead of with the navigation command?

**Comment by diemol** [comment]
I have noticed this and I have chatted with Puja about it, I will send a PR for this in the next couple of days.

**Comment by pujagani** [comment]
> @pujagani if server isn't started, wouldn't the failure happen when starting the session instead of with the navigation command?

Apologies. I should have provided more context. The issue is twofold. 1. Whenever the retries happen, they are unlimited. This I was able to reproduce. This is because maxDuration was removed but the backoff policy was not. So it is a bug that was introduced during my changes in the RetryRequestFilyer but it needs fixing (I will be sending out the fix tomorrow). 2. Having retries everywhere is less than ideal. Diego has good insights on that front. Thank you for driving the effort, Diego. 
A combination of both fixes should help resolve this issue.
